java.io.UncheckedIOException: java.nio.file.NoSuchFileException: /home/gitpod/.cache/jenkins-plugin-modernizer-cli/update-center.json

### Jenkins and plugins versions report

```
java -jar plugin-modernizer-cli/target/jenkins-plugin-modernizer-999999-SNAPSHOT.jar --dry-run --plugin-file plugins.txt  --recipes AddPluginsBom,AddCodeOwner
Picked up JAVA_TOOL_OPTIONS: -XX:+UseContainerSupport -XX:ActiveProcessorCount=1
Starting Plugin Modernizer 
Picked up JAVA_TOOL_OPTIONS: -XX:+UseContainerSupport -XX:ActiveProcessorCount=1
Plugins: [pipeline-milestone-step, mina-sshd-api-core, jakarta-activation-api, github-branch-source, snakeyaml-api, trilead-api, commons-text-api, workflow-multibranch, javax-mail-api, pipeline-stage-tags-metadata, javax-activation-api, plugin-util-api, git-client, scm-api, variant, pipeline-stage-step, font-awesome-api, pipeline-model-extensions, github, build-timeout, pipeline-groovy-lib, configuration-as-code, workflow-support, checks-api, mailer, antisamy-markup-formatter, jackson2-api, pipeline-rest-api, workflow-scm-step, jakarta-mail-api, ws-cleanup, pipeline-model-api, token-macro, pipeline-input-step, metrics, jquery3-api, okhttp-api, credentials-binding, workflow-aggregator, pipeline-build-step, pipeline-stage-view, credentials, mina-sshd-api-common, pipeline-model-definition, caffeine-api, instance-identity, workflow-step-api, locale, plain-credentials, matrix-project, resource-disposer, bouncycastle-api, jaxb, git, bootstrap5-api, workflow-durable-task-step, pipeline-graph-view, ssh-credentials, cloudbees-folder, durable-task, workflow-basic-steps, workflow-job, github-api, timestamper, junit, pipeline-graph-analysis, echarts-api, apache-httpcomponents-client-4-api, structs, ant, workflow-cps, ssh-slaves, sshd, matrix-auth, gradle, jjwt-api, ionicons-api, commons-lang3-api, display-url-api, script-security, branch-api, workflow-api] 
Recipes: [AddPluginsBom, AddCodeOwner] 
GitHub owner: gounthar 
Forking and cloning pipeline-milestone-step locally 
Error extracting repository name:  
java.nio.file.NoSuchFileException: /home/gitpod/.cache/jenkins-plugin-modernizer-cli/update-center.json
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:218)
        at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:484)
        at java.base/java.nio.file.Files.newOutputStream(Files.java:228)
        at java.base/java.nio.file.Files.write(Files.java:3512)
        at java.base/java.nio.file.Files.writeString(Files.java:3734)
        at java.base/java.nio.file.Files.writeString(Files.java:3674)
        at io.jenkins.tools.pluginmodernizer.core.utils.JenkinsPluginInfo.getCachedUpdateCenterData(JenkinsPluginInfo.java:49)
        at io.jenkins.tools.pluginmodernizer.core.utils.JenkinsPluginInfo.extractRepoName(JenkinsPluginInfo.java:26)
        at io.jenkins.tools.pluginmodernizer.core.github.GHService.forkCloneAndCreateBranch(GHService.java:58)
        at io.jenkins.tools.pluginmodernizer.core.impl.PluginModernizer.start(PluginModernizer.java:40)
        at io.jenkins.tools.pluginmodernizer.cli.Main.run(Main.java:132)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2030)
        at picocli.CommandLine.executeHelpRequest(CommandLine.java:2016)
        at picocli.CommandLine.executeHelpRequest(CommandLine.java:1987)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2272)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at io.jenkins.tools.pluginmodernizer.cli.Main.main(Main.java:34)
Failed to process plugin: pipeline-milestone-step 
java.io.UncheckedIOException: java.nio.file.NoSuchFileException: /home/gitpod/.cache/jenkins-plugin-modernizer-cli/update-center.json
        at io.jenkins.tools.pluginmodernizer.core.utils.JenkinsPluginInfo.extractRepoName(JenkinsPluginInfo.java:37)
        at io.jenkins.tools.pluginmodernizer.core.github.GHService.forkCloneAndCreateBranch(GHService.java:58)
        at io.jenkins.tools.pluginmodernizer.core.impl.PluginModernizer.start(PluginModernizer.java:40)
        at io.jenkins.tools.pluginmodernizer.cli.Main.run(Main.java:132)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2030)
        at picocli.CommandLine.executeHelpRequest(CommandLine.java:2016)
        at picocli.CommandLine.executeHelpRequest(CommandLine.java:1987)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2272)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at io.jenkins.tools.pluginmodernizer.cli.Main.main(Main.java:34)
Caused by: java.nio.file.NoSuchFileException: /home/gitpod/.cache/jenkins-plugin-modernizer-cli/update-center.json
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:218)
        at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:484)
        at java.base/java.nio.file.Files.newOutputStream(Files.java:228)
        at java.base/java.nio.file.Files.write(Files.java:3512)
        at java.base/java.nio.file.Files.writeString(Files.java:3734)
        at java.base/java.nio.file.Files.writeString(Files.java:3674)
        at io.jenkins.tools.pluginmodernizer.core.utils.JenkinsPluginInfo.getCachedUpdateCenterData(JenkinsPluginInfo.java:49)
        at io.jenkins.tools.pluginmodernizer.core.utils.JenkinsPluginInfo.extractRepoName(JenkinsPluginInfo.java:26)
        ... 10 common frames omitted
Forking and cloning mina-sshd-api-core locally 
```

### What Operating System are you using (both controller, and any agents involved in the problem)?

GitPod

### Reproduction steps

` java -jar plugin-modernizer-cli/target/jenkins-plugin-modernizer-999999-SNAPSHOT.jar --dry-run --plugin-file plugins.txt  --recipes AddPluginsBom,AddCodeOwner`

### Expected Results

I thought that with the merge of #87 we would be good with the `--plugin-file` option.

### Actual Results

`java.io.UncheckedIOException: java.nio.file.NoSuchFileException: /home/gitpod/.cache/jenkins-plugin-modernizer-cli/update-center.json`

### Anything else?

_No response_

### Are you interested in contributing a fix?

_No response_

--- a/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/CacheManager.java
+++ b/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/CacheManager.java
@@ -1,0 +1,34 @@
+package io.jenkins.tools.pluginmodernizer.core.impl;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import io.jenkins.tools.pluginmodernizer.core.utils.JenkinsPluginInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CacheManager {
+    private static final Logger LOG = LoggerFactory.getLogger(JenkinsPluginInfo.class);
+    private final Path cache;
+
+    public CacheManager(Path cache) {
+        this.cache = cache;
+    }
+
+    void createCache() {
+        if (!Files.exists(cache)) {
+            try {
+                Path parent = cache.getParent();
+                if (parent != null && !Files.exists(parent)) {
+                    Files.createDirectory(parent);
+                }
+                Files.createDirectory(cache);
+                LOG.info("Creating cache at {}", cache);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+    }
+}

--- a/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java
+++ b/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java
@@ -18,10 +18,13 @@ public class PluginModernizer {
 
     private final GHService ghService;
 
+    private final CacheManager cacheManager;
+
     public PluginModernizer(Config config) {
         this.config = config;
         this.mavenInvoker = new MavenInvoker(config);
         this.ghService = new GHService(config);
+        this.cacheManager = new CacheManager(config.getCachePath());
     }
 
     public void start() {
@@ -31,6 +34,9 @@ public class PluginModernizer {
         LOG.debug("Cache Path: {}", config.getCachePath());
         LOG.debug("Dry Run: {}", config.isDryRun());
         LOG.debug("Maven rewrite plugin version: {}", Settings.MAVEN_REWRITE_PLUGIN_VERSION);
+
+        cacheManager.createCache();
+
         for (String plugin : config.getPlugins()) {
             String pluginPath = Settings.TEST_PLUGINS_DIRECTORY + plugin;
             String branchName = "apply-transformation-" + plugin;

--- a/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java
+++ b/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java
@@ -31,6 +31,7 @@ public class PluginModernizer {
         LOG.info("Plugins: {}", config.getPlugins());
         LOG.info("Recipes: {}", config.getRecipes());
         LOG.info("GitHub owner: {}", config.getGithubOwner());
+        LOG.info("Update Center Url: {}", config.getJenkinsUpdateCenter());
         LOG.debug("Cache Path: {}", config.getCachePath());
         LOG.debug("Dry Run: {}", config.isDryRun());
         LOG.debug("Maven rewrite plugin version: {}", Settings.MAVEN_REWRITE_PLUGIN_VERSION);

--- a/plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/impl/CacheManagerTest.java
+++ b/plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/impl/CacheManagerTest.java
@@ -1,15 +1,14 @@
 package io.jenkins.tools.pluginmodernizer.core.impl;
 
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.io.TempDir;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
 public class CacheManagerTest {
 

--- a/plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/impl/CacheManagerTest.java
+++ b/plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/impl/CacheManagerTest.java
@@ -1,0 +1,36 @@
+package io.jenkins.tools.pluginmodernizer.core.impl;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CacheManagerTest {
+
+    @TempDir
+    Path tempDir;
+
+    @Test
+    void testCreateCacheWhenNotExists() throws IOException {
+        Path cachePath = tempDir.resolve("cache");
+        assertFalse(Files.exists(cachePath));
+        CacheManager cacheManager = new CacheManager(cachePath);
+        cacheManager.createCache();
+        assertTrue(Files.exists(cachePath));
+    }
+
+    @Test
+    void testCreateCacheWhenExists() throws IOException {
+        Path cachePath = tempDir.resolve("cache");
+        Files.createDirectories(cachePath);
+        CacheManager cacheManager = new CacheManager(cachePath);
+        cacheManager.createCache();
+        assertTrue(Files.exists(cachePath));
+    }
+}

